feat: enable cal-video-event-type-settings

hemantmm opened the pull request:
closes: #21863

## What does this PR do?

- Fixes #21863 
- Fixes CAL-5942

This PR enables Cal Video settings for event types by:
1. Adding default Cal Video settings when initializing the form
2. Including calVideoSettings in the payload when a Cal Video location is present
3. Explicitly clearing calVideoSettings from the payload when Cal Video location is removed (to prevent stale settings from being submitted)

## Updates since last revision

Addressed review feedback from @cubic-dev-ai:
- Fixed issue where `calVideoSettings` was still being submitted even when the event no longer has a Cal Video location (the property was present from the spread in `filteredPayload`)
- Added explicit `delete filteredPayload.calVideoSettings` when `hasCalVideoLocation` is false
- Removed duplicate logic for setting `calVideoSettings`

## Visual Demo (For contributors especially)

#### Video Demo (if applicable):

Before:

https://github.com/user-attachments/assets/1688ffdf-858b-4c56-b641-da3c65f92f1e


After:

https://github.com/user-attachments/assets/4ac97812-eae3-49af-a8b3-c48a8ddac8ab

Cal.com Atoms testing:

https://github.com/user-attachments/assets/e3ec56c7-ea56-414a-8cd1-caaf2071250e

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

1. Create an event type with Cal Video as a location
2. Save the event type - calVideoSettings should be included in the payload
3. Remove Cal Video location from the event type
4. Save again - calVideoSettings should NOT be included in the payload

## Human Review Checklist

- [ ] Verify that deleting `calVideoSettings` when Cal Video location is removed is the intended behavior
- [ ] Confirm the logic correctly handles the case when calVideoSettings was previously set but Cal Video is no longer a location

## Checklist

- [x] I have read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- [x] My code follows the style guidelines of this project
- [x] I have checked if my changes generate no new warnings

---

Link to Devin run: https://app.devin.ai/sessions/666a456bb2604467ab0a391c14e64184
Requested by: unknown ()

hemantmm committed:
feat: cal-video-event-type-settings

closes: #21863

hemantmm committed:
Merge branch 'main' into cal-video-event-type-settings

hemantmm committed:
Merge branch 'main' into cal-video-event-type-settings

kart1ka committed:
Merge branch 'main' into cal-video-event-type-settings

kart1ka commented on packages/platform/atoms/event-types/hooks/useEventTypeForm.ts:
```diff
@@ -411,7 +424,13 @@ export const useEventTypeForm = ({
+      filteredPayload.calVideoSettings = values.calVideoSettings ?? defaultCalVideoSettings;
+    }
+
+    if (dirtyFieldExists || hasCalVideoLocation) {
```
If cal video is already selected as the location, this makes the save button clickable even when there is no change in the event type settings. This is not the behaviour we want.

kart1ka reviewed (changes requested):
Left a comment. Also the change should be tested in event type settings atom. As per your attached video you have shown the change in event type settings web and not atom. Can you update and test if this works for event settings in atom?

hemantmm committed:
Merge branch 'main' into cal-video-event-type-settings

hemantmm committed:
Merge branch 'main' into cal-video-event-type-settings

hemantmm committed:
Merge branch 'main' into cal-video-event-type-settings

hemantmm committed:
Merge branch 'main' into cal-video-event-type-settings

hemantmm committed:
fix: event type setting

Signed-off-by: Hemant M Mehta <hemant29mehta@gmail.com>

hemantmm committed:
Merge branch 'main' into cal-video-event-type-settings

hemantmm committed:
Merge branch 'cal-video-event-type-settings' of https://github.com/hemantmm/cal.com into cal-video-event-type-settings

hemantmm committed:
Merge branch 'main' into cal-video-event-type-settings

kart1ka commented on .eslintrc.js:
```diff
@@ -28,6 +28,8 @@ module.exports = {
             patterns: [
               {
                 group: ["@calcom/trpc/*", "@trpc/*"],
+                message:
+                  "tRPC imports are blocked in packages/app-store. Move UI to apps/web/components/apps or introduce an API boundary.",
```
looks like unrelated change. please remove it.

kart1ka reviewed (changes requested):
Have you tested if this works for cal.com atoms? I still do not see any loom video that shows this fix works for atoms. Can you please test this and attach a loom video of the same? We do not merge changes without proper video evidence attached.

hemantmm committed:
Merge branch 'main' into cal-video-event-type-settings

hemantmm committed:
fix: duplicate key

Signed-off-by: Hemant M Mehta <hemant29mehta@gmail.com>

hemantmm committed:
Merge branch 'main' into cal-video-event-type-settings

hemantmm committed:
Merge branch 'main' into cal-video-event-type-settings

hemantmm committed:
Merge branch 'main' into cal-video-event-type-settings

hemantmm committed:
Merge branch 'main' into cal-video-event-type-settings

hemantmm committed:
Merge branch 'main' into cal-video-event-type-settings

hemantmm committed:
Merge branch 'main' into cal-video-event-type-settings

hemantmm committed:
Merge branch 'main' into cal-video-event-type-settings

devin-ai-integration[bot] committed:
fix: clear calVideoSettings when Cal Video location is removed

Address review feedback to explicitly delete calVideoSettings from the
payload when the event type no longer has a Cal Video location. This
prevents stale calVideoSettings from being submitted.

Also removed duplicate logic for setting calVideoSettings.

Co-authored-by: hemantmm <hemantmm@users.noreply.github.com>
Co-Authored-By: unknown <>

Ryukemeister committed:
Merge branch 'main' into cal-video-event-type-settings

Ryukemeister reviewed (commented):
hi @hemantmm, thank you for your contribution. however this is not the correct implementation for the fix. the issue mentioned in #21863 is not feasible for atoms, i've explained it in detail there. i'm gonna close this PR for now.
